From 4a09616112de8b9e4920c06b8aa6ae0428cbb5d5 Mon Sep 17 00:00:00 2001
From: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date: Wed, 7 Feb 2024 08:26:34 +0100
Subject: [PATCH] src/pam_python.c: Use new PyConfig API when building against
 Python 3.12 any beyond.

---
 src/pam_python.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/pam_python.c b/src/pam_python.c
index 215e64f..bdb9a0b 100644
--- a/src/pam_python.c
+++ b/src/pam_python.c
@@ -127,6 +127,14 @@ static char libpython_so[]	= LIBPYTHON_SO;
 static void initialise_python(void)
 {
 #if	PY_MAJOR_VERSION*100 + PY_MINOR_VERSION >= 204
+#if	PY_MAJOR_VERSION*100 + PY_MINOR_VERSION >= 312
+  PyConfig config;
+  PyConfig_InitPythonConfig(&config);
+  config.write_bytecode = 1;
+  config.use_environment = 1;		/* Required to mitigate CVE-2019-16729 */
+  config.user_site_directory = 1;	/* Required to mitigate CVE-2019-16729 */
+  config.isolated = 1;
+#else
   Py_DontWriteBytecodeFlag = 1;
   Py_IgnoreEnvironmentFlag = 1;	/* Required to mitigate CVE-2019-16729 */
   Py_NoUserSiteDirectory = 1;	/* Required to mitigate CVE-2019-16729 */
@@ -135,6 +143,7 @@ static void initialise_python(void)
 #endif
   /* Py_NoSiteFlag = 1; 	Breaks too many things */
   Py_InitializeEx(0);
+#endif
 #else
   size_t		signum;
   struct sigaction	oldsigaction[NSIG];
